### PR TITLE
Adding General Development Guidelines

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Summary
Adding General Development Guidelines

### CODE REVIEW

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# General Guidelines
+
+Keep in mind these things when developing terra components
```
Comment (tbiethman) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]:
```suggestion
Keep these things in mind when developing with Terra components:
```

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+* Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, classes, and undocumented side effects.
```
Comment (mjhenkes) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]: Should we mention that css files are internal implementation and not considered part of the api?

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+* Try and keep your DOMs small. If you have a list that could have thousands of items in it, you might need to rethink your design
```
Comment (tbiethman) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]:
```suggestion
* Try to minimize your application's DOM footprint. An application's DOM size is inversely proportional to its performance. If you have a list that could have thousands of items in it, you need to rethink your design.
```

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+* Take some time to learn the basics of JavaScript and React
```
Comment (tbiethman) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]: Links to resources might be helpful

Comment (mjhenkes) [reply src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]: https://developer.mozilla.org/en-US/docs/Web/JavaScript

**Comment by mjhenkes** [comment]
Should we add a comment recommending consumers to extend our web-pack config?

**Comment by ryanthemanuel** [comment]
> Should we add a comment recommending consumers to extend our web-pack config?

Done

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# General Guidelines
+
+Keep these things in mind when developing with Terra components:
+
+* Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, CSS specifics, themeable variables, class names, and undocumented side effects.
```
Comment (yuderekyu) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]:
```suggestion
* Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, CSS specifics, themeable variables, class names, and undocumented side effects. These details are not part of the documented API contract and are subject to change at any time.
```

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# General Guidelines
+
+Keep these things in mind when developing with Terra components:
+
+* Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, CSS specifics, themeable variables, class names, and undocumented side effects. These details are not part of the documented API contract and are subject to change at any time.
+* Try to minimize your application's DOM footprint. An application's DOM size is inversely proportional to its performance. If you have a list that could have thousands of items in it, you need to rethink your design.
+* Take some time to learn the basics of [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) and [React](https://reactjs.org/tutorial/tutorial.html)
```
Comment (mjhenkes) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]:
```suggestion
* Take some time to learn the basics of [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) and [React](https://reactjs.org/tutorial/tutorial.html).
```

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+* Pay attention to your dependencies. Dependencies that are only used in development should be dev dependencies. Make sure all peer dependencies are met.
+* Use Terra's [webpack](https://engineering.cerner.com/terra-ui/getting-started/terra-ui/webpack) configuration where possible
```
Comment (mjhenkes) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]:
```suggestion
* Use Terra's [webpack](https://engineering.cerner.com/terra-ui/getting-started/terra-ui/webpack) configuration where possible.
```

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# General Guidelines
+
+Keep these things in mind when developing with Terra components:
+
+* Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, CSS specifics, themeable variables, class names, and undocumented side effects. These details are not part of the documented API contract and are subject to change at any time.
+* Try to minimize your application's DOM footprint. An application's DOM size is inversely proportional to its performance. If you have a list that could have thousands of items in it, you need to rethink your design.
+* Take some time to learn the basics of [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript) and [React](https://reactjs.org/tutorial/tutorial.html).
+* Pay attention to your dependencies. Dependencies that are only used in development should be dev dependencies. Make sure all peer dependencies are met.
```
Comment (emilyrohrbough) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]: should we add be sure learn about NPM dependency management & link to NPM

File: src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# General Guidelines
```
Comment (emilyrohrbough) [inline src/terra-dev-site/guide/GeneralGuidelines.l.guide.mdx]: Can we move this to come 2nd in the list
```
What is Terra
General Guidelines
How to Install
Webpack etc

```

File: src/terra-dev-site/guide/GeneralGuidelines.b.guide.mdx [hunk 10]
```diff
@@ -3,6 +3,7 @@
 Keep these things in mind when developing with Terra components:
 
 * Terra components are black boxes. You should not rely on internal implementation specifics. This includes DOM structure, CSS specifics, themeable variables, class names, and undocumented side effects. These details are not part of the documented API contract and are subject to change at any time.
+* Javascript components in `priviate` directories are not part of Terras public API and should not be consumed directly.
```
Comment (yuderekyu) [inline src/terra-dev-site/guide/GeneralGuidelines.b.guide.mdx]:
```suggestion
* Javascript components in `private` directories are not part of Terras public API and should not be consumed directly.
```

Comment (mjhenkes) [reply src/terra-dev-site/guide/GeneralGuidelines.b.guide.mdx]: spelling, lol

**Pull request merged**